backend/de.metas.device.api/src/main/java/de/metas/device/api/hook/RunParameters.java

```diff
@@ -44,11 +45,36 @@ public Optional<BigDecimal> getSingleAsBigDecimal(@NonNull final String paramNam
 				.map(value -> NumberUtils.asBigDecimal(value, null));
 	}
 
+	@NonNull
+	public Optional<Integer> getSingleAsIntegerForSuffix(@NonNull final String suffix)
+	{
+		return getSingleForSuffix(suffix)
+				.map(NumberUtils::asInt);
+	}
+
 	@NonNull
 	public Optional<String> getSingle(@NonNull final String paramName)
 	{
 		return Optional.ofNullable(parameters.get(paramName))
 				.filter(list -> list.size() == 1)
 				.map(list -> list.get(0));
 	}
+
+	@NonNull
+	private Optional<String> getSingleForSuffix(@NonNull final String suffix)
```

Comment:
please extract this into a method like `singleElementOrEmpty` in `de.metas.util.collections.CollectionUtils`